keepassxc-browser/background/keepass.js

```diff
@@ -168,18 +169,24 @@ keepass.retrieveCredentials = async function(tab, args = []) {
 
 keepass.generatePassword = async function(tab) {
     if (!keepass.isConnected) {
-        return [];
+        return '';
+    }
+
+    if (keepass.passwordRequestSent) {
+        return undefined;
```

Comment:
Technically we check for this condition on the keepassxc side and just focus the existing password generator. I kind of like that behavior.

Reply:
I don't like this either, but then we'll have to accept that with Firefox some messages are left in the buffer until browser is closed. I don't know why, but sometimes no messages are returned from KeePassXC.
Also, there's a possibility that this boolean stays false if some error occurs in the extension, so I'm removing this. Let's add some kind of checking for this later if actually needed.

Reply:
Plus I need to modify the return value to an Object so we can properly identify when an error message is needed to show.